Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift

```diff
@@ -0,0 +1,119 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if compiler(>=5.5) && canImport(_Concurrency)
+import NIOCore
+import NIOHTTP1
+
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+struct HTTPClientResponse {
+    var version: HTTPVersion
+    var status: HTTPResponseStatus
+    var headers: HTTPHeaders
+    var body: Body
+
+    struct Body {
+        private let bag: Transaction
+        private let reference: ResponseRef
+
+        fileprivate init(_ transaction: Transaction) {
+            self.bag = transaction
+            self.reference = ResponseRef(transaction: transaction)
+        }
+    }
+
+    init(
+        bag: Transaction,
+        version: HTTPVersion,
+        status: HTTPResponseStatus,
+        headers: HTTPHeaders
+    ) {
+        self.body = .init(bag)
```

Comment:
Nit: I this this is less clear and less concise than

```suggestion
        self.body = Body(bag)
```